File: app/src/main/java/com/wangwh/posterdemo/models/Location.java
package com.wangwh.posterdemo.models;

/**
 * @Author: 王文辉
 * @Date: 2022-04-19 14:17
 * @Desc: 坐标属性
 * @Note:
 */
public class Location {
    private int x;
    private int y;
    /**
     * 基于的原点
     * @see OriginType
     */
    private int origin;

    public OriginType getOriginType() {
        return OriginType.Companion.find(origin);
    }

    public int getX() {
        return x;
    }

    public void setX(int x) {
        this.x = x;
    }

    public int getY() {
        return y;
    }

    public void setY(int y) {
        this.y = y;
    }

    public int getOrigin() {
        return origin;
    }

    public void setOrigin(int origin) {
        this.origin = origin;
    }
}

File: app/src/main/java/com/wangwh/posterdemo/models/Location.kt
package com.wangwh.posterdemo.models

import androidx.annotation.Keep

/**
 * @Author: 王文辉
 * @Date: 2022-04-19 09:31
 * @Desc: 坐标
 * @Note:
 */
@Keep
open class Location(
    val x: Int,
    val y: Int,
    val origin: Int = 0
) {
    val originType: OriginType get() = OriginType.find(origin)
}